src/app.ts

```diff
@@ -267,25 +214,36 @@ const balena = BalenaSdk({
 
 (async () => {
 	try {
-		let ipAddr: string;
-		// Get IP address for the specified interface, and the TLD to use.
+		const ipAddrs: string[] = [];
 		if (process.env.INTERFACE) {
-			ipAddr = getNamedInterfaceAddr(process.env.INTERFACE);
+			// get the first non link-local IP address for the specified interface for each address family
+			ipAddrs.push(
+				getAddrs4(process.env.INTERFACE).flatMap((f) =>
+					f ? f.address : [],
+				)[0],
+				getAddrs6(process.env.INTERFACE).flatMap((f) =>
+					f ? f.address : [],
+				)[0],
```

Comment:
there could be interfaces with only local addresses, so compacting makes sense and let's us get rid of the extra flatMap
```suggestion
				..._.compact( [
				getAddrs4(process.env.INTERFACE)[0],
				getAddrs6(process.env.INTERFACE)[0]
				]),
```

Reply:
similar change needed below of course